docs/async/async-csharp-vb.rst

```diff
@@ -0,0 +1,200 @@
+Async Programming in C#/VB
+==========================
+By `Phillip Carter`_
+
+Async programming is traditionally associated with callbacks and a general lack of straightforward code.  One might even say that straightforward code and scalable code are often at odds with one another.  This is not a problem with the C#/VB async model!  It is a language-level construct which hides complexity and allows you to focus on the problem at hand without making the code confusing in the process.
+
+The core of this are the ``async`` and ``await`` keywords (``Async`` and ``Await`` in VB), which allow for simple changes to code with a huge impact.  For example:
+
+.. code-block:: c#
+
+	button.Clicked += async (o, e) =>
+	{
+	    var data = await _httpClient.DownloadStringAsync(URL);
+	    DisplayData(data);
+	};
+	
+And that's it!  No callbacks and no library to wrestle with.
+
+For those who are more theoretically-inclined, the model used by C# and VB is an implementation of the `Future and Promise concurrency model <https://en.wikipedia.org/wiki/Futures_and_promises>`_.
+
+A few important things to know before continuing:
+
+* Async code relies on ``Task<T>`` and ``Task``, which are constructs used to abstract the notion of a job that you need to wait on for a result.  They also act as the return types, except for the case of Event Handlers (these return ``void``).  `More on Task and Task<T>`_.
+* Marking a method as ``async`` tells the compiler that you're expecting to ``await`` the completion of a task.   Application of ``await`` is optional, but recommended in most cases.
+* ``await`` can only be used inside an async method.
+* When the ``await`` keyword is applied, it suspends the calling method and returns control back to its caller until the awaited task is complete.  This is what makes it responsive.
```

Comment:
"responsive" is a client concept, while async/await is equally applicable to the server.